src/csync/csync_private.h

```diff
@@ -67,14 +67,44 @@ enum csync_replica_e {
   REMOTE_REPLICA
 };
 
+
+/*
+ * This is a structurere similar to QStringRef
+ * The difference is that it keeps the QByteArray by value and not by pointer
+ * And it only implements a very small subset of the API that is required by csync, the API can be
+ * added as we need it.
+ */
+class ByteArrayRef
+{
+    QByteArray _arr;
+    int _begin = 0;
+    int _size = -1;
+
+public:
+    ByteArrayRef(const QByteArray &arr = {}, int begin = 0, int size = -1)
+        : _arr(arr)
+        , _begin(begin)
+        , _size(qMin(arr.size() - begin, size < 0 ? arr.size() : size))
```

Comment:
Could this be an assert instead?

Reply:
I did not make it an assert after all,  this is also how QByteArray::left works, by being very tollerent in what we accept